students_proj/src/main/java/com/musala/simple/students/Main.java

```diff
@@ -0,0 +1,279 @@
+	 * Validates the command line input arguments.
+	 *
+	 * @param args  A String array with command line arguments
+	 * @return true if the first argument in the array
+	 * is a valid path to a file with extension ".json".
+	 */
+	private static boolean inputIsValid(String[] args) {
+
+		/* Check if any arguments are provided */
+		if (args.length == 0) {
+			System.err.println(ERROR_PATH_MISSING);
+			return false;
+		}
+
+		String argPath = args[0];
+		File file = new File(argPath);
+
+		/* Check if path exists */
+		if (!file.exists()) {
+			System.err.println(ERROR_INVALID_PATH);
+
+			return false;
+		}
+
+		/* Check if the valid path is a file */
+		if (!file.isFile()) {
+			System.err.println(ERROR_NOT_A_FILE);
+
+			return false;
+		}
+
+		/* Check the file extension to be JSON. */
+		if (!argPath.endsWith(JSON_FILE_EXTENSION)) {
+			System.err.println(ERROR_FILE_NOT_JSON);
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Creates a backup Json file using the students objects
+	 * provided as a Json String parameter. If a backup file
+	 * already exists, it's being overwritten.
+	 *
+	 * @param studentsJsonInfo A Json String, containing the students
+	 * to be written to the backup file.
+	 */
+	private static void createBackupJson(String studentsJsonInfo) {
+		try (FileWriter fileWriter = new FileWriter(BACKUP_JSON_PATH, false)) {
+			fileWriter.write(studentsJsonInfo);
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+
+	/**
+	 * Reads the backup Json file (if exists) and prints the
+	 * students information from it.
+	 *
+	 */
+	private static void readBackupJson() throws ResourceDoesNotExistException {
+		File backupFile = new File(BACKUP_JSON_PATH);
+
+		if (!backupFile.exists()) {
+			throw new ResourceDoesNotExistException(ERROR_NO_BACKUP_JSON);
+		}
+		String backupStudentsJson = readFile(BACKUP_JSON_PATH);
+
+		StudentWrapper studentWrapper = new Gson().fromJson(backupStudentsJson, StudentWrapper.class);
+		printStudents(Arrays.asList(studentWrapper.students));
+	}
+
+	/**
+	 * Updates the backup file by adding the new students. If the backup
+	 * file does not exist the {@link Main#createBackupJson(String)} is called
+	 * and the backup is created with the new students.
+	 *
+	 * @param studentsJsonInfo A Json String, containing the students
+	 * to be added to the backup file.
+	 */
+	public static void updateBackupFile(String studentsJsonInfo) {
+		File backupFile = new File(BACKUP_JSON_PATH);
+
+		if (!backupFile.exists()) {
+			System.out.println(INFO_CREATING_BACKUP);
+			createBackupJson(studentsJsonInfo);
+		} else {
+			String backupFileInfo = readFile(BACKUP_JSON_PATH);
+			Student[] backupStudents = new Gson().fromJson(backupFileInfo, StudentWrapper.class).students;
+			Student[] newStudents = new Gson().fromJson(studentsJsonInfo, StudentWrapper.class).students;
+			
+			StudentGroup sGroup = new StudentGroup();
+			sGroup.fillStudentGroup(backupStudents);
+			sGroup.fillStudentGroup(newStudents);
```

Comment:
In relation to my comment for not using the constructor `public FileWriter(File file, boolean append)` - did you intentionally set append to be false because of the consecutive calling ot these two statements?
```java
sGroup.fillStudentGroup(backupStudents);
sGroup.fillStudentGroup(newStudents);
```

Reply:
No. I think I'm actually using the `public FileWriter(String fileName, boolean append)` constructor, not the `public FileWriter(File file, boolean append)` , because I thought the `public FileWriter(String fileName)` would naturally append to the file, if it already exists. And I wanted to always create a new file. Now I see I can achieve the same thing with the  `public FileWriter(String fileName)`. :)